app/src/main/java/com/android/sample/model/recipe/RecipeRepository.kt

```diff
@@ -0,0 +1,6 @@
+package com.android.sample.model.recipe
+
+interface RecipeRepository {
+
+  fun random(nbOfElements: Long, onSuccess: (List<Recipe>) -> Unit, onFailure: (Exception) -> Unit)
```

Comment:
You should change the type of the nbOfElements parameter from Long to Int because it is not necessary, an Int is enough for this project.